lib/rails_translation_manager/locale_checker/all_locales.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+class AllLocales
+  attr_reader :locale_path
+
+  def initialize(locale_path)
+    @locale_path = locale_path
+  end
+
+  def generate
+    paths = locale_file_paths.compact
+
+    raise NoLocaleFilesFound, "No locale files found for the supplied locale path" if paths.blank?
+
+    paths.flat_map do |locale_group|
+      {
+        locale: locale_group[:locale],
+        keys: all_keys_for_locale(locale_group)
+      }
+    end
+  end
+
+  private
+
+  def locale_file_paths
+    I18n.available_locales.map do |locale|
+      grouped_paths = Dir[locale_path].select do |path|
+        locale = locale.downcase
+
+        path =~ %r{/#{locale}/} || path =~ %r{/#{locale}\.yml$}
+      end
+
+      { locale: locale.downcase, paths: grouped_paths } if grouped_paths.present?
+    end
+  end
+
+  def all_keys_for_locale(locale_group)
+    locale_group[:paths].flat_map do |path|
+      get_keys_for_locale(locale_file_path: path)
+    end
+  end
+
+  def get_keys_for_locale(hsh: nil, key_chain: nil, ary: [], locale_file_path: nil)
+    hsh ||= YAML.load_file(locale_file_path)
+    keys = hsh.keys
+    keys.each do |key|
+      if hsh.fetch(key).is_a?(Hash)
+        get_keys_for_locale(hsh: hsh.fetch(key), key_chain: "#{key_chain}.#{key}", ary: ary)
+      else
+        keys.each do |final_key|
+          ary << "#{key_chain}.#{final_key}"
+        end
+      end
+    end
+
+    # remove locale prefix from keys, e.g: ".en.browse.page" -> "browse.page"
```

Comment:
I found this comment helpful 👍 I feel like the `IncompatiblePlurals` class's methods could do with some similar comments which give examples of inputs and outputs. I was only vaguely following along [here](https://github.com/alphagov/rails_translation_manager/pull/12/commits/790076117cd32b25293a8403012489cca6e0d671#diff-667e6893a3236e587c2311d7b3a359c2d91bb229535bb1862b7b8245524f31b1R54-R76)!

Reply:
Ah, good idea! I've added some more example comments to `IncompatiblePlurals` which should highlight what the tricky methods are doing.

Reply:
Thanks. I'm still struggling a bit with the `get_keys_for_locale` method, I think it's the implicit return I'm not sure about. I'm not sure whether it can be refactored though.

Reply:
It is quite hard to understand (and refactor 😄)! I'll have a think and see if I can refactor, at a future point?